dbclient/JobsClient.py

```diff
@@ -143,6 +148,8 @@ def adjust_ids_for_cluster(settings): #job_settings or task_settings
         with open(jobs_log, 'r') as fp:
             for line in fp:
                 job_conf = json.loads(line)
+                if 'job_id' in job_conf and checkpoint_job_configs_set.contains(str(job_conf['job_id'])):
```

Comment:
need to do this `str(...)`  because otherwise the job_id is recognized as integer which becomes str vs int which never matches. (in which case, the checkpoint never recognizes that the job_id is already checkpointed)

Reply:
@kevinkim9264 I noticed something similar happened with object/directory IDs when doing comparisons in my pruning script... is there anywhere else that this might happen?

Reply:
@gregwood-db can you explain the "something similar" happened part?
I just checked the checkpointed files for "import_notebooks.log" file and and they are all String. 
This would happen if the id we checkpoint is pure integer

Reply:
@kevinkim9264 I think it'll will helpful to add your explanation about str() to code comments.